nilearn/datasets/tests/test_utils.py

```diff
@@ -364,6 +364,19 @@ def test_uncompress():
 
         shutil.rmtree(dtemp)
 
+        dtemp = mkdtemp()
```

Comment:
if you got the energy do you think that you could refactor this test into 3 tests:
- one for tar
- one for gz
- one for tgz

not sure we can properly parametrize it.